app/client/lib/collections.js

```diff
@@ -3,26 +3,31 @@
 // we use {connection: null} to prevent them from syncing with our not existing Meteor server
 
 
+
 Wallets = new Mongo.Collection('wallets', {connection: null});
 new PersistentMinimongo2(Wallets, 'ethereum_wallet');
+Wallets = new NetworkInfo.ProxyCollection(Wallets);
 
 CustomContracts = new Mongo.Collection('custom-contracts', {connection: null});
 new PersistentMinimongo2(CustomContracts, 'ethereum_wallet');
+CustomContracts = new NetworkInfo.ProxyCollection(CustomContracts);
 
 // Contains the transactions
 Transactions = new Mongo.Collection('transactions', {connection: null});
 new PersistentMinimongo2(Transactions, 'ethereum_wallet');
+Transactions = new NetworkInfo.ProxyCollection(Transactions);
 
 // Contains the pending confirmations
 PendingConfirmations = new Mongo.Collection('pending-confirmations', {connection: null});
 new PersistentMinimongo2(PendingConfirmations, 'ethereum_wallet');
+PendingConfirmations = new NetworkInfo.ProxyCollection(PendingConfirmations);
 
 // Contains the custom contract events
 Events = new Mongo.Collection('events', {connection: null});
 new PersistentMinimongo2(Events, 'ethereum_wallet');
+Events = new NetworkInfo.ProxyCollection(Events);
 
 // Contains Coin Information
 Tokens = new Mongo.Collection('tokens', {connection: null});
```

Comment:
I see this is the only place that actually initialises the old collections with all the data. Maybe this would be a good place to run `_.each(Tokens.find({network:{$exists: false}}, {sort:{symbol:1}}), function(e,i){` and then run a `web3.eth.getCode` on each token and custom contract without a network assigned.

This way data migration could be done incrementally, every time the user connects to a different network it would try to identify more data to migrate.